scaleway/helpers_vpc.go

```diff
@@ -40,3 +44,60 @@ func vpcAPI(m interface{}) (*vpc.API, error) {
 
 	return vpc.NewAPI(meta.scwClient), nil
 }
+
+const minVPCTimeOut = 10 * time.Second
+const PollVPCTimeOut = 30 * time.Second
+
+// retryVPCContext is a basic wrapper around StateChangeConf that will just retry with a poll interval
+// a function until it no longer returns an error.
+//
+// Cancellation from the passed in context will propagate through to the
+// underlying StateChangeConf
+func retryVPCContext(ctx context.Context, timeout time.Duration, f resource.RetryFunc) error {
+	// These are used to pull the error out of the function; need a mutex to
+	// avoid a data race.
+	var resultErr error
+	var resultErrMu sync.Mutex
+
+	c := &resource.StateChangeConf{
+		Pending:      []string{"retryableerror"},
+		Target:       []string{"success"},
+		Timeout:      timeout,
+		MinTimeout:   minVPCTimeOut,
+		PollInterval: PollVPCTimeOut,
+		Refresh: func() (interface{}, string, error) {
+			rErr := f()
+
+			resultErrMu.Lock()
+			defer resultErrMu.Unlock()
+
+			if rErr == nil {
+				resultErr = nil
+				return 42, "success", nil
```

Comment:
Why 42? Could we have something meaningful?

Reply:
I've took the inspiration from the terraform helpers.  The difference here that we can handle the poll intervals as the timeouts parameters.